app/src/layout/MissionHall.tsx

```diff
@@ -135,20 +135,107 @@ export default function MissionHall({ userData }: MissionHallProps) {
             const capped = isErrand
               ? errandsLeft <= 0
               : isMedical
-                ? userData.dailyMedicalMissions >= MEDICAL_MISSIONS_PER_DAY ||
-                  filtered.length === 0
+                ? userData.dailyMedicalMissions >= MEDICAL_MISSIONS_PER_DAY
                 : userData.dailyMissions >= MISSIONS_PER_DAY;
             // Checks
             const rankCheck = availableUserRanks.includes(setting.rank) || isErrand;
             const medicalCheck = isMedical
-              ? (medicalRanks?.some(
-                  (q) =>
-                    q.questRank === setting.rank &&
-                    (q.medicalRank === "NONE" || q.medicalRank === userMedicalRank),
-                ) ?? false)
+              ? filtered.length > 0  // Show colored if ANY medical missions are available
               : true;
             const grayScale = count === 0 || capped || !rankCheck || !medicalCheck;
 
+
+            if (isMedical) {
+              return (
```

Comment:
Since we have this same general logic for A-rank, I think we should generalize it out, so that we can re-use it also for other missions more easily in the future if needed, and maintain easier